### ISSUE TITLE
Loading DMF devices (version 0.3.0) broken for some 0.2.0 files (Trac #60)

### ISSUE THREAD

**Issue by cfobel** [wheeler-microfluidics/microdrop#64]
Test device at 'microdrop/tests/devices/device 0 v0.2.0' fails to load with the following error:

Traceback (most recent call last):
  File "C:\Python\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "D:\docs\ryan\dev\python\microdrop\microdrop\tests\test_dmf_device.py", line 26, in load_device
    DmfDevice.load(name)
  File "microdrop\dmf_device.py", line 118, in load
    out._upgrade()
  File "microdrop\dmf_device.py", line 164, in _upgrade
    float(l['y'])) for l in e.path] + ['z'])
KeyError: '\'x\'\n-------------------- >> begin captured logging << --------------------\nroot: DEBUG: [DmfDevice].load("D:\\docs\\ryan\\dev\\python\\microdrop\\microdrop\\tests\\devices\\device 0 v0.2.0")\nroot: INFO: Loading DmfDevice from D:\\docs\\ryan\\dev\\python\\microdrop\\microdrop\\tests\\devices\\device 0 v0.2.0\nroot: DEBUG: Loaded object from pickle.\nroot: DEBUG: [DmfDevice]._upgrade()\nroot: DEBUG: [DmfDevice] version=0.2.0, class_version=0.3.0\n--------------------- >> end captured logging << ---------------------'

Migrated from http://microfluidics.utoronto.ca/ticket/60
```json
{
    "status": "closed", 
    "changetime": "2014-04-17T19:39:01", 
    "description": "Test device at 'microdrop/tests/devices/device 0 v0.2.0' fails to load with the following error:\n\nTraceback (most recent call last):\n  File \"C:\\Python\\lib\\site-packages\\nose\\case.py\", line 197, in runTest\n    self.test(*self.arg)\n  File \"D:\\docs\\ryan\\dev\\python\\microdrop\\microdrop\\tests\\test_dmf_device.py\", line 26, in load_device\n    DmfDevice.load(name)\n  File \"microdrop\\dmf_device.py\", line 118, in load\n    out._upgrade()\n  File \"microdrop\\dmf_device.py\", line 164, in _upgrade\n    float(l['y'])) for l in e.path] + ['z'])\nKeyError: '\\'x\\'\\n-------------------- >> begin captured logging << --------------------\\nroot: DEBUG: [DmfDevice].load(\"D:\\\\docs\\\\ryan\\\\dev\\\\python\\\\microdrop\\\\microdrop\\\\tests\\\\devices\\\\device 0 v0.2.0\")\\nroot: INFO: Loading DmfDevice from D:\\\\docs\\\\ryan\\\\dev\\\\python\\\\microdrop\\\\microdrop\\\\tests\\\\devices\\\\device 0 v0.2.0\\nroot: DEBUG: Loaded object from pickle.\\nroot: DEBUG: [DmfDevice]._upgrade()\\nroot: DEBUG: [DmfDevice] version=0.2.0, class_version=0.3.0\\n--------------------- >> end captured logging << ---------------------'", 
    "reporter": "ryan", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1397763541728826", 
    "component": "microdrop", 
    "summary": "Loading DMF devices (version 0.3.0) broken for some 0.2.0 files", 
    "priority": "major", 
    "keywords": "", 
    "version": "0.1", 
    "time": "2012-02-24T15:01:12", 
    "milestone": "Microdrop 1.0", 
    "owner": "cfobel", 
    "type": "defect"
}
```

**Comment by cfobel** [comment]
Trac update at 20120224T18:02:36: 
* christian@fobel.net commented:

> commit d65bb06584c0620164acbcd7f4948338ecc9072c
> Author: Christian Fobel <christian@fobel.net>
> Date:   Fri Feb 24 13:00:20 2012 -0500
> 
>     Add support for 'Z' cmd in device 0.2.0->0.3...
>     
>     Add support for 'Z' command in device 0.2.0->0.3.0 upgrade.  Fixes #64.
>     Add upgraded v0.3.0 version of device 0 in tests/devices.


* christian@fobel.net changed resolution from "" to "fixed"
* christian@fobel.net changed status from "new" to "closed"

**Comment by cfobel** [comment]
Trac update at 20140417T19:39:01: ryan changed component from "core" to "microdrop"
